pkg/utils/kubernetes/networking/services.go

```diff
@@ -45,3 +48,92 @@ func WaitForServiceLoadBalancerAddress(ctx context.Context, c *kubernetes.Client
 		}
 	}
 }
+
+// WaitForServicePortConnected waits for a service provided by namespace/name
+// being able to connect by TCP. It creates a LoadBalancer with same selector and ports,
+// and try to establish TCP connection to the loadbalacer IP and port.
```

Comment:
```suggestion
// WaitForServicePortConnected waits until it can make successful TCP connections
// to a service (provided by namespace/name). This will temporarily create a LoadBalancer
// type Service to allow connections to the Service and port from outside the cluster while
// the connection attempts are made using the LoadBalancer public address.
```

Reply:
updated.